### ISSUE TITLE
LPub3D LDView preferences have no effect

### ISSUE THREAD

**Issue by Landcross** [trevorsandy/lpub3d#98]
### Subject
I just updated to LPub3D and it contains some cool features and enhancements. However, I also noticed a big downgrade in render-quality, which is something I didn't like to see since I'm in the midst of working on a project.

Then I noticed that LPub3D now has a LDView render preferences dialog of its own instead of using the one from within LDView itself. I copied my settings (as much as possible, see my other issue) and hit apply and OK. I re-rendered the page. Nothing changed. I restarted LPub3D. Nothing changed.

### Environment
* Version of LPub3D - 2.3.0 Revision 0 Build 887
* Operating system - Windows 10 Pro x64

I can share logs and/or configuration files if necessary, just let me know which ones you need. I'm currently in the midst of working on a project. I think I'm gonna downgrade back to LPub3D 2.2.x as this is a major downgrade for me personally.

**Comment by trevorsandy** [comment]
Thanks again for sharing this information. Most likely, if your image quality is not the same, as before, it is because the renderer is not receiving the same configuration parameters is it previously did. You can confirm this by reviewing the log files the see precisely what parameters and .ini file is being sent to the renderer.

It could be helpful to see the following:

- The LPub3D log 
- Your LDView.ini file settings
- An example of the reduction is image quality - for example before and after
- A copy of the registry entries

There is a much simpler way to evaluate the new capabilities in Windows' LPub3D than 'updating' and, consequently 'downgrading' etc. Simply download and extract a completely contained portable distribution and immediately begin editing your instructions with the new capabilities. Furthermore, the portable distribution now contains the registry entries, in addition to system and user data. The registry file is hosted under the 'config' subfolder of the extracted distribution.

Cheers,

**Comment by Landcross** [comment]
The problem is not that the image quality is different. The cause of that is quite clear: the new included LDView preferences are still at default, which differs from my standalone LDView preferences.  The problem, is that I can't seem to change the preferences of the included LDView. Or rather, that the changes don't have any effect on the outputted images.

I followed your advice and took a look at the LDView.ini file. Or rather, files. I noticed that LPub3D has created 2 folders under AppData -> Local. One is called _LPub3D_ and the other is called _LPub3D Software_. The folder LPub3D contains the executable and everything else. The folder LPub3D Software contains only a single folder called LPub3D which then contains the following folders:

- 3rdParty
- cache
- extras
- libraries
- logs

Curiously, these same exact folders exists in AppData -> Local -> LPub3D. They also both contain (partly) the same contents. Which in this case also means that there are 2 ldview.ini files.

In any case, I don't know if that information is of any use for this issue.

When editing the preferences through the UI in LPub3D the .ini file(s) don't change I think. When I edit the .ini file in the LPub3D Software folder **manually** and refresh the render in LPub3D the image does change, though the setting I want to change (line thickness) doesn't seem to change. But maybe I'm adding wrong values in the .ini file.

So, in short, it seems like the LDView Preferences dialog in LPub3D does not change the preferences. But it's a bit late here already and it's been a long day. I can do some more testing tomorrow.

**Comment by trevorsandy** [comment]
Corrected.
